add some documentation (#7)
MatrixCore/MatrixCore: Sources/MatrixClient/MatrixClient.docc/Resolve-homeserver-url.md

> # Resolve Homeserver url

Reviewer: ```suggestion
# Resolve Homeserver URL
```

> Resolve the actuall home server url from the given well known url

Reviewer: ```suggestion
Resolve the actual homeserver URL from the given well-known url
```

## Overview

> With ``MatrixHomeserver`` you can resolve an homeserver url usable by ``MatrixClient``.

Reviewer: ```suggestion
With ``MatrixHomeserver`` you can resolve a homeserver URL usable by ``MatrixClient``.
```

## Resolving
To resolve the homeserver url you need to init ``MatrixHomeserver`` with

> ``MatrixHomeserver/init(resolve:withUrlSession:)``.

Reviewer: ```suggestion
``MatrixHomeserver.init(resolve:withUrlSession:)``.
```

```swift

> let homesrver = try await MatrixHomeserver(resolve: "https://matrix.org/")

Reviewer: ```suggestion
let homeserver = try await MatrixHomeserver(resolve: "https://matrix.org/")
```
